javaslang/src/main/java/javaslang/control/Try.java

```diff
@@ -719,6 +719,35 @@ default void orElseRun(Consumer<? super Throwable> action) {
         return f.apply(this);
     }
 
+    /**
+     * Provides try's finally behavior no matter what the result of the operation is.
+     *
+     * @param runnable A runnable
+     * @return this {@code Try}.
+     * @throws NullPointerException if {@code runnable} is null
+     */
+    default Try<T> andFinally(Runnable runnable) {
+        Objects.requireNonNull(runnable, "runnable is null");
+        return andFinallyTry(runnable::run);
+    }
+
+    /**
+     * Provides try's finally behavior no matter what the result of the operation is.
+     *
+     * @param runnable A runnable
+     * @return this {@code Try}.
+     * @throws NullPointerException if {@code runnable} is null
+     */
+    default Try<T> andFinallyTry(CheckedRunnable runnable) {
+        Objects.requireNonNull(runnable, "runnable is null");
+        try {
+            runnable.run();
+            return this;
+        } catch (Throwable t) {
+            return new Failure<>(t);
```

Comment:
@vicmosin can you provide a test please? I've overseen it - I'm on a mobile phone...